src/internet_message.c

```diff
@@ -163,22 +160,26 @@ static int im_header_add_fold_ctx(struct im_header_context *ctx)
  */
 int im_header_feed(struct im_header_context *ctx, char c)
 {
+	jsval header;
+
 	switch (ctx->state) {
 	case IM_H_NAME1:
 		if (strchr(tab_space, c)) {
-			if (list_empty(ctx->hdrs))
```

Comment:
I think you're missing a case here. The original algorithm idea was this:
* If the line starts with whitespace when we're expecting a new header, it means that we're actually parsing an additional folding to the previous header.
* However, if there is no previous header, it's a syntax error.

I'm not really sure how you detect this case of syntax error in the new implementation.